Social login and Websocket feature

abdulfousan opened the pull request:
-Social authentication added
-login page shows cancel in username label : bug fixed
-Usertracker menu added
-Websocket functionality added
-User tracker page added
-Sometimes in auth.jwt.service.js returns bearerToken null in
authenticateSuccess function : fixed
-stylesheet updated
-Readme updated for web socket

abdulfousan committed:
Social login and Websocket feature

-Social authentication added
-login page shows cancel in username label : bug fixed
-Usertracker menu added
-Websocket functionality added
-User tracker page added
-Sometimes in auth.jwt.service.js returns bearerToken null in
authenticateSuccess function : fixed
-stylesheet updated
-Readme updated for web socket

ruddell commented:
Wow, thank you!  I'll test this today but it looks good from a glance through the code.

Why was every other file's 'File Mode' changed?  That is the only issue I see so far.

abdulfousan commented:
You're welcome

I don't know why file mode changed, may be its my ide issue I guess..anyway will find a way to fix soon..

abdulfousan committed:
File mode Fix

abdulfousan committed:
Revert "File mode Fix"

This reverts commit f88af5b22fe8a662e0ebe60a12b42a1cd7ff8517.

abdulfousan committed:
File Mode Changed - Fixed

previously after commit some file mode has changed from chmod 100644 to
100755 now its fixed

ruddell commented on generators/app/index.js:
```diff
@@ -198,7 +198,10 @@ module.exports = yeoman.Base.extend({
-
+    
+     fse.copySync(this.jhipsterHome+ '/src/main/webapp/app/admin/admin.state.js','./app/main/jhipster/admin/admin.state.js');
+     fse.copySync(this.jhipsterHome+ '/src/main/webapp/app/admin/tracker','./app/main/jhipster/admin/tracker');
```
Add `if (this.enableWebsocket) {` around files specific to projects with websockets

ruddell commented on generators/app/index.js:
```diff
@@ -239,7 +242,8 @@ module.exports = yeoman.Base.extend({
       copyTemplate('custom/account/_reset.finish.html', 'app/main/jhipster/account/reset/finish/reset.finish.html', 'stripHtml', this, {}, true);
-
+      //admin template
+      copyTemplate('custom/admin/_tracker.html', 'app/main/jhipster/admin/tracker/tracker.html', 'stripHtml', this, {}, true);
```
Add `if (this.enableWebsocket) {` here too

ruddell commented on generators/app/index.js:
```diff
@@ -276,8 +280,11 @@ module.exports = yeoman.Base.extend({
       this.template('jhipster/_profile.service.js', 'app/main/jhipster/services/profiles/profile.service.js');
-
-
+      //social login fix
```
Use `this.enableSocialSignIn` around the social files and `this.enableWebsocket` around the tracker service

ruddell commented on generators/app/templates/custom/_login.html:
```diff
@@ -1,36 +1,38 @@
-  </ion-content>
+        <a class="alert-link" href="" ng-click="vm.requestResetPassword()" translate="login.password.forgot">Forgot Password?</a>
+        <br />
+        <a class="alert-link" href="" ng-click="vm.register()" translate="global.messages.info.register">Register a new account</a>
```
Use <% if (enableSocialSignIn) { %> around the social buttons

ruddell commented on generators/app/templates/custom/_menu.html:
```diff
@@ -1,59 +1,62 @@
+                <ion-item ng-switch-when="true" menu-close ui-sref="password">
+                    <span translate="global.menu.account.password">Password</span>
+                </ion-item>
+                <ion-item ng-switch-when="true" has-authority="ROLE_ADMIN" menu-close ui-sref="jhi-tracker">
```
Websocket file here too

ruddell commented on generators/app/templates/custom/admin/_tracker.html:
```diff
@@ -0,0 +1,17 @@
+<ion-view view-title="{{'tracker.title'| translate}}">
```
Make this line look like [this](https://github.com/ruddell/generator-jhipster-ionic/blob/fa3f9b49fad00f3ed0dfeb89a49738651ec7b1fa/generators/app/templates/custom/account/_settings.html#L1).  The translate regex doesn't pick this up currently

ruddell commented:
I think I got everything, sorry for all of the comments! Thanks again, this is great work.

ruddell commented on generators/app/templates/custom/_login.html:
```diff
@@ -1,36 +1,38 @@
-    <div class="list">
-      <label class="item item-input">
-        <span class="input-label" translate="entity.action.cancel">Login</span>
-        <input type="text" class="form-control" id="username" placeholder="{{'global.form.username.placeholder' | translate}}"
```
We need the space before the `| translate` so that the translate regex can remove it for apps without i18n.  Also please add the space back in the password's placeholder.

abdulfousan committed:
Bug Fixes

-Added this.enableWebsocket and this.enableSocialSignIn
-Added the space in the password's placeholder.

abdulfousan committed:
Fix Choosing project path errors without trailing / (#6)

abdulfousan committed:
Bug Fix - i18n directory not being copied into android asset (#7)

bug fixed based on @rpelissier patch.Reference
https://github.com/ruddell/generator-jhipster-ionic/issues/7#issuecommen
t-223028915

ruddell commented:
Please make a new pull request for any other changes (use branches like [JHipster](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) does).
